packages/js/tsconfig-cjs.json

```diff
@@ -1,7 +1,7 @@
 {
 	"extends": "./tsconfig",
 	"compilerOptions": {
-		"target": "es6",
+		"target": "esnext",
```

Comment:
This is quite the jump, not that I am opposed. But will this cause any issues for the newest release of packages, for the plugins that bundle it within their code?
We did probably have to create major release versions for each new package or not?

Reply:
Although it may be fine anyhow given the `module` hasn't changed - my `tsconfig` knowledge is quite limited.

Reply:
Ultimately the goal of the `target` option is to tell TypeScript what features the output JavaScript should be able to make use of. On the browser front, Gutenberg's `tsconfig` files also target `"esnext"` and their Babel configuration targets browsers that WordPress has declared compatibility for.

With that said, however, I think you're right that this is a bit heavy-handed. I'll change this to `"es2019"` which is the same as our `tsconfig.json`. This is supported by the version of Node we've declared compatibility with, as well as all of the browsers that we are compatible with. Given that our ESM builds already target this I don't think it requires a major version release.